products/endpoints/api.yaml

```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(nelsonjr): Make all Zone and Region resource ref
+
+--- !ruby/object:Api::Product
+name: Endpoints
+display_name: Endpoints Service
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://servicemanagement.googleapis.com/v1/
```

Comment:
So one thing that appears to have been generated is the endpoints custom endpoint, which is a bit tricky since the existing endpoints_service resource uses the service management custom endpoint. I think it would be better if there were just one custom endpoint for the endpoints_service resource and its associated IAM resources.

Reply:
So change this to the service management product, and override the name via legacy name?

Reply:
It's definitely going to need some thought on how to do it well. One option would be to create a function to determine the custom endpoint name, which would return a custom_endpoint_name value set in terraform.yaml, and default to whatever we currently use.

Reply:
Renamed the product to ServiceManagement and then overrode the name of the IAM resources to match the existing handwritten resource